File: app/src/main/java/cc/smartcash/wallet/Models/WebWalletUserAvailableRequest.java

package cc.smartcash.wallet.Models;

import android.os.Parcel;
import android.os.Parcelable;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "data"
})
public class WebWalletUserAvailableRequest implements Serializable, Parcelable {

    public final static Parcelable.Creator<WebWalletUserAvailableRequest> CREATOR = new Creator<WebWalletUserAvailableRequest>() {


        @SuppressWarnings({
                "unchecked"
        })
        public WebWalletUserAvailableRequest createFromParcel(Parcel in) {
            return new WebWalletUserAvailableRequest(in);
        }

        public WebWalletUserAvailableRequest[] newArray(int size) {
            return (new WebWalletUserAvailableRequest[size]);
        }

    };
    private final static long serialVersionUID = -1418443063754402134L;
    @JsonProperty("data")
    @SerializedName("data")
    @Expose
    private String data;

    public WebWalletUserAvailableRequest(String data) {
        setData(data);
    }

    protected WebWalletUserAvailableRequest(Parcel in) {
        this.data = ((String) in.readValue((String.class.getClassLoader())));
    }

    public WebWalletUserAvailableRequest() {
    }

    @JsonProperty("data")
    public String getData() {
        return data;
    }

    @JsonProperty("data")
    public void setData(String data) {
        this.data = data;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(data);
    }

    public int describeContents() {
        return 0;
    }

}

File: app/src/main/java/cc/smartcash/wallet/Models/WebWalletUserAvailableRequest.kt
package cc.smartcash.wallet.Models

import android.os.Parcel
import android.os.Parcelable

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("data")
class WebWalletUserAvailableRequest : Serializable, Parcelable {
    @JsonProperty("data")
    @SerializedName("data")
    @Expose
    @get:JsonProperty("data")
    @set:JsonProperty("data")
    var data: String? = null

    constructor()

    constructor(data: String) {
        this.data = data
    }

    private constructor(`in`: Parcel) {
        this.data = `in`.readValue(String::class.java.classLoader) as String
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(data)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<WebWalletUserAvailableRequest> {
        override fun createFromParcel(parcel: Parcel): WebWalletUserAvailableRequest {
            return WebWalletUserAvailableRequest(parcel)
        }

        override fun newArray(size: Int): Array<WebWalletUserAvailableRequest?> {
            return arrayOfNulls(size)
        }
    }


}